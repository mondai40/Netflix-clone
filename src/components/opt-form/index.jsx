import React from 'react';

import { Container, Input, Button, Text, Break } from './styles/opt-form';

export default function OptFlorm({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

OptFlorm.Input = function OptFormInput({ ...restProps }) {
  return <Input {...restProps} />;
};

OptFlorm.Button = function OptFormButton({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try Now" />
    </Button>
  );
};

OptFlorm.Text = function OptFormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

OptFlorm.Break = function OptFormBreak({ ...restProps }) {
  return <Break {...restProps} />;
};
