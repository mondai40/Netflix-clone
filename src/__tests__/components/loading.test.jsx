import React from 'react';
import { render } from '@testing-library/react';

import { Loading } from '../../components';

describe('<Loading />', () => {
  it('renders the <Loading /> component', () => {
    const { container, getByTestId } = render(
      <Loading src="/imges/karl.png" data-testid="loading" />
    );
    expect(getByTestId('loading')).toBeTruthy();
    expect(getByTestId('lodaing-picute')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Loading.ReleaseBody />', () => {
    const { container } = render(<Loading.ReleaseBody />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
