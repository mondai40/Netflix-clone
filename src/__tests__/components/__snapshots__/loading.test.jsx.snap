// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders the <Loading /> component 1`] = `
<div
  class="loading__Spinner-sc-11yhc25-0 jdEyAt"
  data-testid="loading"
>
  <img
    class="loading__Picture-sc-11yhc25-1 siMrc"
    data-testid="lodaing-picute"
    src="/images/users//imges/karl.png.png"
  />
</div>
`;

exports[`<Loading /> renders the <Loading.ReleaseBody /> 1`] = `null`;
